SimPEG/directives.py

```diff
@@ -1173,3 +1406,264 @@ def endIter(self):
+            phi_d.append(dmis(self.opt.xc))
+
+        phi_m = []
+        for reg in self.reg.objfcts:
+            phi_m.append(reg(self.opt.xc))
+
+        ### pass attributes values to invProb
+        ### Assume last reg.objfct is the coupling
+        self.invProb.phi_d_joint = phi_d
+        self.invProb.phi_m_joint = phi_m[:-1]
+        self.invProb.phi_c = phi_m[-1]
+        self.invProb.betas = self.reg.multipliers[:-1]
+        self.invProb.lambd = self.reg.multipliers[-1]
+
+
+class SaveOutputEveryIteration_Joint(SaveEveryIteration, InversionDirective):
+    '''
+    SaveOutputEveryIteration for Joint Inversions.
+    Saves information on the tradeoff parameters, data misfits, regularizations,
+    coupling term, number of CG iterations, and value of cost function.
+    '''
+    header = None
+    save_txt = True
+    betas = None
+    phi_d = None
+    phi_m = None
+    phi_c = None
+    phi = None
+
+    def initialize(self):
+        if self.save_txt is True:
+            print(
+                "SimPEG.SaveOutputEveryIteration will save your inversion "
+                "progress as: '###-{0!s}.txt'".format(self.fileName)
+            )
+            f = open(self.fileName+'.txt', 'w')
+            self.header = "  #          betas            lambda         joint_phi_d                joint_phi_m            phi_c       iterCG     phi    \n"
+            f.write(self.header)
+            f.close()
+
+        # Create a list of each
+        self.betas = []
+        self.lambd = []
+        self.phi_d = []
+        self.phi_m = []
+        self.phi = []
+        self.phi_c = []
+
+    def endIter(self):
+
+        self.betas.append(["{:.2e}".format(elem) for elem in self.invProb.betas])
+        self.phi_d.append(["{:.3e}".format(elem) for elem in self.invProb.phi_d_joint])
+        self.phi_m.append(["{:.3e}".format(elem) for elem in self.invProb.phi_m_joint])
+        self.lambd.append("{:.2e}".format(self.invProb.lambd))
+        self.phi_c.append(self.invProb.phi_c)
+        self.phi.append(self.opt.f)
+
+        if self.save_txt:
+            f = open(self.fileName+'.txt', 'a')
+            f.write(
+                ' {0:2d}  {1}  {2}  {3}  {4}  {5:1.4e}  {6:d}  {7:1.4e}\n'.format(
+                self.opt.iter,
+                self.betas[self.opt.iter-1],
+                self.lambd[self.opt.iter-1],
+                self.phi_d[self.opt.iter-1],
+                self.phi_m[self.opt.iter-1],
+                self.phi_c[self.opt.iter-1],
+                self.opt.cg_count,
+                self.phi[self.opt.iter-1] )
+            )
+            f.close()
+
+    def load_results(self):
+        results = np.loadtxt(self.fileName+str(".txt"), comments="#")
+        self.betas = results[:, 1]
+        self.lambd = results[:, 2]
+        self.phi_d = results[:, 3]
+        self.phi_m = results[:, 4]
+        self.phi_c = results[:, 5]
+        self.f = results[:, 7]
+
+class BetaEstimate_ByEig_Joint(InversionDirective):
```

Comment:
For more details, you could also check how the new beta_estimate is set up:
https://github.com/simpeg/simpeg/blob/96959904ecb75529353d6d23230d1be996a9b14b/SimPEG/directives.py#L195